parsec/data.c

```diff
@@ -560,3 +560,54 @@ parsec_data_destroy( parsec_data_t *data )
 #endif
     PARSEC_OBJ_RELEASE(data);
 }
+
+
+/**
+ * Arena-datatype management.
+ */
+
+/* adt constructor is split in two stages: when the adt is constructed
+ * with OBJ_CONSTRUCT (or allocated with OBJ_NEW), we initialize the
+ * adt object with static zero values.
+ * The user code must then call the parameterized constructor
+ * (below) that attaches the datatype and creates the associated arena.
+ *
+ * When the adt object is destructed (or released), the implicit destructor
+ * releases all resources created by both split constructor stages.
+ *
+ * Freeing the type is not part of the implicit destructor.
+ */
+int parsec_arena_datatype_construct(parsec_arena_datatype_t *adt,
+                                    size_t elem_size,
+                                    size_t alignment,
+                                    parsec_datatype_t opaque_dtt) {
```

Comment:
Ok, the braces are widely inconsistent in the existing code but I'll modify to not add more of these then. 

I don't think the init name is more clear, especially given that it is user-visible, unlike the implicit constructor. Normally we use init to initialize frameworks, and construct to, well, construct static objects. We do have examples of using init, initialize, create, and I'm sure some other nomenclatures because we are not very consistent :( I'll rename construct_zero construct, and recheck what is best for the stage2 constructor.

Reply:
You're talking about the internal _CONSTRUCT macro, but this is different there are no static objects here. All our matrices have this capability under the _init moniker.